libcudacxx/include/cuda/std/__type_traits/disjunction.h

```diff
@@ -51,26 +51,24 @@ struct _OrImpl<false>
 template <class... _Args>
 using _Or _LIBCUDACXX_NODEBUG_TYPE = typename _OrImpl<sizeof...(_Args) != 0>::template _Result<false_type, _Args...>;
 
-#if _CCCL_STD_VER > 2011
-
-#  ifdef _CCCL_COMPILER_MSVC
+#ifdef _CCCL_COMPILER_MSVC
 template <class... _Args>
 struct disjunction : false_type
 {};
 
 template <class _First, class... _Rest>
 struct disjunction<_First, _Rest...> : _OrImpl<true>::template _Result<false_type, _First, _Rest...>
 {};
-#  else
+#else
 template <class... _Args>
 struct disjunction : _Or<_Args...>
 {};
-#  endif // !MSVC
+#endif // !MSVC
 
+#if _CCCL_STD_VER >= 2014 && !defined(_LIBCUDACXX_HAS_NO_VARIABLE_TEMPLATES)
```

Comment:
How can we be on C++14, but have no variable templates?

Reply:
In this case it is old gcc compilers